### PR TITLE
Fix duplicate export properties

### PR DESCRIPTION
**Pull request by mjriley**
## Summary
Initial ticket: https://dimagi-dev.atlassian.net/browse/SAAS-11729.

In rare circumstances, duplicate export properties can be generated.  This occurs because we do not account for the possibility of two properties, with the same path but different doc types.  Bizarrely, despite `closed` being a reserved keyword, several of the applications on the domain in the above ticket generate a `closed` property -- including applications which appear to be blank. This closed property gets added like a user-specified property, with the doc type `ScalarItem`. `closed` then also is generated for the internal properties, this time using the type `ExportItem`. Because of the difference, `TableConfiguration.get_column()` returns as if no column exists. This causes the export generator to continually add on new columns (first you'll have 2, then the next edit with have 4, then 6, etc).

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

A test suite for the above scenario is covered within _test_table_configuration.py_

### QA Plan

No QA is required. While I would like to be sure that removing the cache regeneration code won't cause any problems, I don't believe QA would be able to test that, as the comment listed it as an edge case anyway.

### Safety story
Beyond the test suite, I verified this locally. The previous optimizations were put in place because [the case export edit form failed to load for large cases](https://dimagi-dev.atlassian.net/browse/SAAS-10103). I wanted to be sure that these changes would not re-introduce this problem, so I imported the `irc-erd-kenya` applications locally, and looked at the timings before and after these changes.

Before, generating the case export took ~21 seconds. After, likely due to the removal of some redundant code, I'm seeing the export take ~20.6 seconds. While I still had the redundant code in, I saw times rise by ~.5 secs, which makes sense, as the changes do perform more computation. Given the difference between the two, I think it's very unlikely that these changes make any difference performance-wise. However, 20 seconds is still far too long for this process, and I hope to tackle this is another ticket I'll attach later.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by orangejenny** [comment]
Haven't reviewed this, just noting that this

> Bizarrely, despite closed being a reserved keyword, several of the applications on the domain in the above ticket generate a closed property -- including applications which appear to be blank.

might be because `closed` hasn't been a reserved keyword for very long, and was added due to problems in that same project, see https://github.com/dimagi/commcare-hq/pull/28885

**Comment by mjriley** [comment]
Thanks for the link -- I did see that closed was added recently, but it's still surprising to see `closed` as a property for blank applications that have not even been built yet. But, perhaps that was the case before it got added as reserved (I should check that)

File: corehq/apps/export/tests/test_table_configuration.py [hunk 1]
```diff
@@ -12,6 +13,24 @@
...
+            columns=[
+                ExportColumn(item=prop1), ExportColumn(item=prop2)
+            ]
+        )
+
+        scalarIndex, _ = table_config.get_column(path, 'ScalarItem', None)
+        exportIndex, _ = table_config.get_column(path, 'ExportItem', None)
+
+        self.assertEqual(scalarIndex, 0)
+        self.assertEqual(exportIndex, 1)
```
Comment (millerdev) [inline corehq/apps/export/tests/test_table_configuration.py:31]: After reading the description of this PR (and knowing the context from pairing with you on this bug), I expected this test to verify that there can only be a single column with a given path. However, it looks like this is verifying that the duplicate items, both having the same path, exist in separate columns. In other words, this test seems like it is ensuring that duplicates can exist when I expected it to verify the opposite. I'm confused. What did I miss?

File: corehq/apps/export/models/new.py [hunk 2]
```diff
@@ -590,6 +594,15 @@ def _create_index(_path, _transform):
         for index in indices:
             column = self.columns[index]
 
+            # The column cache can currently get out of sync because other code directly modifies
+            # the columns. For example, ExportInstance._move_selected_columns_to_top just completely
+            # overwrites column order without updating our cache.
+            # TODO: Modify code such that all table manipulation is done through this class
```
Comment (millerdev) [inline corehq/apps/export/models/new.py:600]: :+1:

**Review by millerdev — COMMENTED** [review]
It doesn't feel right to approve this since I am apparently not understanding the solution. However, I'm sure you know what you're doing here, so I will be happy to approve promptly once my misunderstanding is cleared up.

**Review by millerdev — APPROVED** [review]
Offlined with Matt and he clarified my misunderstanding, which was the definition of a "duplicate" in this context. It is legal for two columns with different doc types to have the same path. These are not considered duplicates, which was my misunderstanding.

This is solving a different "duplicate" problem where extra copies of properties having matching paths were added each time an export was saved because the column lookup for those properties returned `None, None`. This PR makes the number of properties with matching paths remain constant rather than doubling on each save. The test verifies that the root cause of the bug is no longer present: properties with matching paths can now be referenced as expected.

**Pull request merged**